packages/beeq/src/components/date-picker/bq-date-picker.tsx

```diff
@@ -0,0 +1,544 @@
+
+  /** The select input value represents the currently selected date or range and can be used to reset the field to a previous value.
+   * All dates are expected in ISO-8601 format (YYYY-MM-DD). */
+  @Prop({ reflect: true, mutable: true }) value: string;
+
+  // Prop lifecycle events
+  // =======================
+
+  @Watch('value')
+  handleValueChange() {
+    if (Array.isArray(this.value)) {
+      this.hasValue = this.value.some((val) => val.length > 0);
+      return;
+    }
+
+    this.hasValue = isDefined(this.value);
+  }
+
+  @Watch('type')
+  checkPropValues() {
+    validatePropValue(DATE_PICKER_TYPE, 'single', this.el, 'type');
+  }
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  /** Callback handler emitted when the input loses focus */
+  @Event() bqBlur!: EventEmitter<HTMLBqDatePickerElement>;
+
+  /**
+   * Callback handler emitted when the input value has changed and the input loses focus.
+   * This handler is called whenever the user finishes typing or pasting text into the input field and then clicks outside of the input field.
+   */
+  @Event() bqChange!: EventEmitter<{ value: string; el: HTMLBqDatePickerElement }>;
+
+  /** Callback handler emitted when the input value has been cleared */
+  @Event() bqClear!: EventEmitter<HTMLBqDatePickerElement>;
+
+  /** Callback handler emitted when the input has received focus */
+  @Event() bqFocus!: EventEmitter<HTMLBqDatePickerElement>;
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  componentDidLoad() {
+    this.handleValueChange();
+  }
+
+  // Listeners
+  // ==============
+
+  @Listen('bqOpen', { capture: true })
+  handleOpenChange(ev: CustomEvent<{ open: boolean }>) {
+    if (!ev.composedPath().includes(this.el)) return;
+
+    this.open = ev.detail.open;
+  }
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  /**
+   * Clears the selected value.
+   *
+   * @return {Promise<void>}
+   * @memberof BqInput
+   */
+  @Method()
+  async clear(): Promise<void> {
+    if (this.disabled) return;
+
+    this.value = undefined;
+
+    this.bqClear.emit(this.el);
+  }
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private handleBlur = () => {
+    if (this.disabled) return;
+
+    this.bqBlur.emit(this.el);
+  };
+
+  private handleFocus = () => {
+    if (this.disabled) return;
+
+    this.bqFocus.emit(this.el);
+  };
+
+  private handleInput = (ev: Event) => {
+    if (this.disabled) return;
+
+    if (!isHTMLElement(ev.target, 'input')) return;
+    this.value = ev.target.value;
+  };
```

Comment:
Since we drop `bqInput` event emitter, we can also remove this method and its usage.